packages/cli/src/util/runner.ts

```diff
@@ -12,25 +12,28 @@ export function runPrismAndSetupWatcher(createPrism: Function, options: CreatePr
         server.fastify.log.info('Restarting Prism...');
 
         getHttpOperationsFromResource(spec)
-          .then(async operations => {
+          .then(operations => {
             if (operations.length === 0) {
               server.fastify.log.info(
                 'No operations found in the current file, continuing with the previously loaded spec.',
               );
             } else {
-              await server.fastify.close();
+              return server.fastify.close()
+                .then(() => {
+                  server.fastify.log.info('Loading the updated operations...');
```

Comment:
Let's try to limit the indentation and leverage the chaining

```ts
export function runPrismAndSetupWatcher(createPrism: Function, options: CreatePrismOptions, spec: string) {
  return createPrism(options).then((server: IPrismHttpServer) => {
    if (server) {
      const watcher = chokidar.watch(spec, { awaitWriteFinish: { stabilityThreshold: 500, pollInterval: 100 } });

      watcher.on('change', () => {
        server.fastify.log.info('Restarting Prism...');

        getHttpOperationsFromResource(spec)
          .then(operations => {
            if (operations.length === 0) {
              server.fastify.log.info(
                'No operations found in the current file, continuing with the previously loaded spec.',
              );
            } else {
              return server.fastify.close().then(() => operations);
            }
          }).then(operations => {
            server.fastify.log.info('Loading the updated operations...');

            return createPrism({ ...options, operations });
          })
          .then(newServer => server = newServer)
          .catch(() => {
            server.fastify.log.info('Something went terribly wrong, trying to start Prism with the original document.');

            return server.fastify
              .close()
              .then(() => createPrism(options))
              .catch(() => process.exit(1))
          });
      });
    }
  });
}
```

Reply:
hm.. @XVincentX, I think this won't work because of:
```
if (operations.length === 0) {
  server.fastify.log.info(
    'No operations found in the current file, continuing with the previously loaded spec.',
  );
}
```
so then, in `}).then(operations => {`, `operations` is undefined